### DISCUSSION TITLE
FileNotFoundError: Unable to open KiCad Schematic Library File device

### DISCUSSION THREAD

**Discussion by Lakroft** [xesscorp/skidl#211] [Q&A] [answered]
I'm a new both in Kicad and Skidl. First thing I've tried to build is an example found in internet, but get "FileNotFoundError: Unable to open KiCad Schematic Library File device" error. My OS is macOS 14.5 (got the same issue on windows 10), KiCad ver. 6.0.11 downloaded from https://downloads.kicad.org/kicad/macos/explore/stable (the same was with KiCad 8, later found a discussion suggested to use v6), Skidl v1.2.1 from pip. I've configured a KICAD_SYMBOL_DIR and checked 'Diode_SMD:D_0201_0603Metric' is exist. 

Did I forget to configure something or it is a bug? 

![Screenshot 2024-05-27 at 14 40 54](https://github.com/devbisme/skidl/assets/12113483/f1b005a0-9797-491b-844f-b9655d941714)
Following is a content of `leds.py`:
```# https://www.baldengineer.com/6-things-i-learned-about-kicad-at-kicon-2019.html/skidl-side-by-side-example#main
from skidl import *

anodes = Bus('a', 6)
cathodes = Bus('k', 16)

led = Part('device', 'D', footprint='Diode_SMD:D_0201_0603Metric', dest=TEMPLATE)

for a in anodes[1:4]:
    for k in cathodes[1:15]:
        led()['A', 'K'] += a, k

for i in range(2,6):
    for k in cathodes[i:i+10:5]:
        led()['A', 'K'] += anodes[5], k

ERC()
generate_netlist()```

**Comment by devbisme** [comment, marked as answer]
Hi. Thanks for trying SKiDL!

First, make sure the KiCad V6 symbol directory is in `lib_search_paths`. Here's what mine looks like:
```python
>>> print(lib_search_paths)
{'spice': ['.'], 'kicad': ['.', '/home/devb/projects/KiCad/kicad_symbols/v6'], 'skidl': ['.', '/home/devb/.pyenv/versions/skidltest/lib/python3.11/site-packages/skidl/libs']}
```
You can see the local directory where my KiCad V6 symbols are stored in the list under the `kicad` key.

If that's OK, then the problem is the need to be a little more explicit with your library name. Try this:
```python
led = Part('Device.kicad_sym', 'D', footprint='Diode_SMD:D_0201_0603Metric', dest=TEMPLATE)
```

Let me know if that works.

(The next release of SKiDL won't require the `kicad_sym` library extension to load modern symbol libraries.)

**Reply by Lakroft** [reply]
```led = Part('Device.kicad_sym', 'D', footprint='Diode_SMD:D_0201_0603Metric', dest=TEMPLATE)```
This one solved the issue. Thank you so much!
